db/storage.go

```diff
@@ -0,0 +1,91 @@
+package db
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/luciodesimone/golang-bootcamp/beer"
+	"io/ioutil"
+	"os"
+	"sync"
+)
+
+//RWMap contains the database structure and locks to prevent data
+//dependency on concurrent access
+type RWMap struct {
+	sync.RWMutex
+	m map[string]beer.Beer
+}
+
+type storage struct {
+	db         *RWMap
+	DB         *RWMap
+	connString string
+	connection *os.File
+}
+
+//New creates a storage with the
+func New(connString string, connection *os.File) DB {
+	return &storage{
+		db:         &RWMap{m: make(map[string]beer.Beer)},
+		connString: connString,
+		connection: connection,
+	}
+}
+
+//Open creates or open an existent file and uploads to memory all the content if it's valid json
+func Open(connStr string) (DB, error) {
+	f, err := os.OpenFile(connStr, os.O_APPEND|os.O_CREATE, os.ModePerm)
```

Comment:
When is this file closed?

I'm trying to think in the event the program panics, for example, what would happen?